risc0/zkvm/sdk/rust/src/receipt.rs

```diff
@@ -88,6 +99,13 @@ impl Receipt {
         self.verify_with_hal(&hal, method_id)
     }
 
+    /// This function is called by [verify](Receipt::verify), which provides the
+    /// CPU HAL generated using the program circuit.
+    ///
+    /// # Arguments
+    ///
+    /// * `hal` - the HAL used to represent the guest program circuit.
```

Comment:
Actually the HAL is independent of the circuit, it's the "Hardware Abstraction Layer" which provides a way to execute portions of the verification in an efficient way based on whether the verify is happening on the host or in the guest (aka recursion).